### ISSUE TITLE
podman stop command doesn't stop systemd services in the container gracefully.

### ISSUE THREAD

**Issue by knaka888** [containers/podman#11140]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

When I execute "podman stop" command for a container where several systemd services are running, the services don't stop gracefully. 

For example, while "postgresql-12" service is running in "postgresql" container, when I execute "systemd stop postgresql-12" inside the container, /var/log/messages includes the following log lines.
Aug  5 09:02:52 postgresql systemd[1]: Stopping PostgreSQL 12 database server...
Aug  5 09:02:52 postgresql systemd[1]: Stopped PostgreSQL 12 database server.

However when I execute "podman stop postgresql" command and then "podman start postgresql" and check /var/log/messages, the above log lines do not appear around the time when "podman stop postgresql" was executed. It looks like the service is killed suddenly. 

I understand that when we execute "podman stop [container]", SIGRTMIN+3 is thrown to the container. But it seems that the container does not handle the signal appropriately.

**Steps to reproduce the issue:**

1. Execute "systemd stop [service]" inside a container and confirm "Stopping [service]" and "Stopped [service]" appear in /var/log/messages.

2. Execute "podman stop [container]" and then "podman start [container]".

3. Check the /var/log/messages inside the container.

**Describe the results you received:**

Log lines of "Stopping [service]" and "Stopped [service]" don't appear around the time when "podman stop [container]" was executed in the /var/log/messages .

**Describe the results you expected:**

Log lines of "Stopping [service]" and "Stopped [service]" should appear around the time when "podman stop [container]" was executed in the /var/log/messages .

**Additional information you deem important (e.g. issue happens only occasionally):**

This issue always happens. 

**Output of `podman version`:**

```
Version:      3.2.3
API Version:  3.2.3
Go Version:   go1.15.13
Built:        Wed Jul 21 13:07:51 2021
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.21.3
  cgroupControllers:
  - cpuset
  - cpu
  - cpuacct
  - blkio
  - memory
  - devices
  - freezer
  - net_cls
  - perf_event
  - net_prio
  - hugetlb
  - pids
  - rdma
  cgroupManager: systemd
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.29-1.el8.3.7.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.29, commit: '
  cpus: 1
  distribution:
    distribution: '"centos"'
    version: "8"
  eventLogger: journald
  hostname: localhost.localdomain
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 4.18.0-305.10.2.el8_4.x86_64
  linkmode: dynamic
  memFree: 207568896
  memTotal: 3918450688
  ociRuntime:
    name: crun
    package: crun-0.20.1-1.el8.3.4.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.20.1
      commit: 0d42f1109fd73548f44b01b3e84d04a279e99d2e
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 1719660544
  swapTotal: 1719660544
  uptime: 1h 16m 18.2s (Approximately 0.04 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 5
    paused: 0
    running: 1
    stopped: 4
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageStore:
    number: 61
  runRoot: /var/run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 3.2.3
  Built: 1626840471
  BuiltTime: Wed Jul 21 13:07:51 2021
  GitCommit: ""
  GoVersion: go1.15.13
  OsArch: linux/amd64
  Version: 3.2.3
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-3.2.3-1.el8.1.3.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

I confirmed this issue on CentOS 8.4.2105 on a VM using VirtualBox 6.1.26.

**Comment by Luap99** [comment]
Can you paste your systemd unit. `podman stop` has a default timeout of 10 seconds before it sends SIGKILL. You can increase the timeout with the `--time` option.

**Comment by mheon** [comment]
I'm assuming that systemd is running inside the container?

A `podman inspect` of the container in question would also help - want to verify that systemd mode is active and the correct stop signal has been set.

We should be sending a special signal (I honestly forget which - I think it's `SIGRTMIN+2` but not 100% sure) to systemd in the container, which causes it to shut down, which should include taking down everything it manages in the container in an orderly fashion. I'd expect to see log messages from systemd to this effect.

**Comment by knaka888** [comment]
Yes, I make use of systemd which is running inside the container.

The systemd unit files I tested are default service unit files of PostgreSQL and OpenSSH services. Both services seem to be 
stopped suddenly without "systemdctl stop [service]" called for each service before they are killed. Therefore I assume that all the systemd services inside a container would have the same results. Though I added "--time 30" option for "podman stop" command, the results were the same.

Here is the output of "podman inspect" command. As for the special signal, I can see **"StopSignal": 37** in the output. So I think the stop signal is **SIGRTMIN+3**. But I don't know why systemd doesn't work well, when "podman stop [container]" is executed.

```
[
    {
        "Id": "3562c3f0ccc272e1b5953c8a2bfa192d81b8abe7a611d3a1055cffcaadaf76f5",
        "Created": "2020-08-27T11:05:29.112881812+09:00",
        "Path": "/sbin/init",
        "Args": [
            "/sbin/init"
        ],
        "State": {
            "OciVersion": "1.0.2-dev",
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 5677,
            "ConmonPid": 5667,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2021-08-06T08:45:19.106348448+09:00",
            "FinishedAt": "2021-08-06T08:43:58.933310213+09:00",
            "Healthcheck": {
                "Status": "",
                "FailingStreak": 0,
                "Log": null
            }
        },
        "Image": "67e9e92360f14cbc0e83a589b28a4ceddd7ad5406d93582db20009108a3d4809",
        "ImageName": "localhost/local/c8-postgresql:latest",
        "Rootfs": "",
        "Pod": "",
        "ResolvConfPath": "/var/run/containers/storage/overlay-containers/3562c3f0ccc272e1b5953c8a2bfa192d81b8abe7a611d3a1055cffcaadaf76f5/userdata/resolv.conf",
        "HostnamePath": "/var/run/containers/storage/overlay-containers/3562c3f0ccc272e1b5953c8a2bfa192d81b8abe7a611d3a1055cffcaadaf76f5/userdata/hostname",
        "HostsPath": "/var/run/containers/storage/overlay-containers/3562c3f0ccc272e1b5953c8a2bfa192d81b8abe7a611d3a1055cffcaadaf76f5/userdata/hosts",
        "StaticDir": "/var/lib/containers/storage/overlay-containers/3562c3f0ccc272e1b5953c8a2bfa192d81b8abe7a611d3a1055cffcaadaf76f5/userdata",
        "OCIConfigPath": "/var/lib/containers/storage/overlay-containers/3562c3f0ccc272e1b5953c8a2bfa192d81b8abe7a611d3a1055cffcaadaf76f5/userdata/config.json",
        "OCIRuntime": "runc",
        "ConmonPidFile": "/var/run/containers/storage/overlay-containers/3562c3f0ccc272e1b5953c8a2bfa192d81b8abe7a611d3a1055cffcaadaf76f5/userdata/conmon.pid",
        "PidFile": "",
        "Name": "postgresql",
        "RestartCount": 0,
        "Driver": "overlay",
        "MountLabel": "system_u:object_r:container_file_t:s0:c441,c474",
        "ProcessLabel": "system_u:system_r:container_init_t:s0:c441,c474",
        "AppArmorProfile": "",
        "EffectiveCaps": [
            "CAP_AUDIT_WRITE",
            "CAP_CHOWN",
            "CAP_DAC_OVERRIDE",
            "CAP_FOWNER",
            "CAP_FSETID",
            "CAP_KILL",
            "CAP_MKNOD",
            "CAP_NET_BIND_SERVICE",
            "CAP_NET_RAW",
            "CAP_SETFCAP",
            "CAP_SETGID",
            "CAP_SETPCAP",
            "CAP_SETUID",
            "CAP_SYS_CHROOT"
        ],
        "BoundingCaps": [
            "CAP_AUDIT_WRITE",
            "CAP_CHOWN",
            "CAP_DAC_OVERRIDE",
            "CAP_FOWNER",
            "CAP_FSETID",
            "CAP_KILL",
            "CAP_MKNOD",
            "CAP_NET_BIND_SERVICE",
            "CAP_NET_RAW",
            "CAP_SETFCAP",
            "CAP_SETGID",
            "CAP_SETPCAP",
            "CAP_SETUID",
            "CAP_SYS_CHROOT"
        ],
        "ExecIDs": [
            "7dfdb550d920c8c1713e1a998e89a1df899f5c793cf96184de4d303d46fb5a0c"
        ],
        "GraphDriver": {
            "Name": "overlay",
            "Data": {
                "LowerDir": "/var/lib/containers/storage/overlay/35e7aa766131d37acbdf2586cb1e6864479aba1ff390fc59a7cba8c5fc5574b6/diff:/var/lib/containers/storage/overlay/019f02550a5a6df893d7f3173720e3d7f138df230ffba2d3fb03576a4d9ae400/diff:/var/lib/containers/storage/overlay/5f0d0572ba17744b44e5c08c80f403dcb52843f5ba362d56910b26c44a9e81fc/diff:/var/lib/containers/storage/overlay/ef8c3cf0b4627b4b08ed1f9269a88de4e59ab9c5d453052dc27e35725cc70b4f/diff:/var/lib/containers/storage/overlay/75a88be88778a13ec85c09c51ab456333b10e922604be727b1136f972be15faf/diff:/var/lib/containers/storage/overlay/bfa474c90d50457d6de0906e5f782c04bd0a2cf681c08af41252c03a51c9df2b/diff:/var/lib/containers/storage/overlay/17784107baa7ad6f35fe78688648e14bf552bbdc63d354320d66524ad2e734ae/diff:/var/lib/containers/storage/overlay/c17a4c303eef9b52c01ef969b71aa1e9e8aeb25a818b2ba734e2129121799015/diff:/var/lib/containers/storage/overlay/cecbdc9c45dc4f386a9f467920c9b21030b9e3e2562c1442da8cbe67a03955f4/diff:/var/lib/containers/storage/overlay/10ec8250f5ec707b37c968e6aa9cacc2834c803b69f32d909bac53fbd1d41706/diff:/var/lib/containers/storage/overlay/5ec3f14f7c5212951d4897defec419d7c3afd5f384455f2b23cfae5ac078f315/diff:/var/lib/containers/storage/overlay/54edd5a63ce474d6538c3ee6aee77db6e52e37b3862feb5a93b8f40fb53c2569/diff:/var/lib/containers/storage/overlay/4d330ab6e8a980ae73947d118271068b3ebc6cdc23571fcecf469243fdbd00e0/diff",
                "MergedDir": "/var/lib/containers/storage/overlay/a006b6faac56ad23da09af05e0d19d31f6e476f4fabbc13068081892151148b7/merged",
                "UpperDir": "/var/lib/containers/storage/overlay/a006b6faac56ad23da09af05e0d19d31f6e476f4fabbc13068081892151148b7/diff",
                "WorkDir": "/var/lib/containers/storage/overlay/a006b6faac56ad23da09af05e0d19d31f6e476f4fabbc13068081892151148b7/work"
            }
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/root/work",
                "Destination": "/tmp/work",
                "Driver": "",
                "Mode": "",
                "Options": [
                    "rbind"
                ],
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
        "Dependencies": [],
        "NetworkSettings": {
            "EndpointID": "",
            "Gateway": "",
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "MacAddress": "",
            "Bridge": "",
            "SandboxID": "",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "22/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "2224"
                    }
                ],
                "5432/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "15432"
                    }
                ]
            },
            "SandboxKey": "/run/netns/cni-2d468f99-0ff0-6e72-d3e3-5772bcf253c1",
            "Networks": {
                "demo-network": {
                    "EndpointID": "",
                    "Gateway": "10.200.1.1",
                    "IPAddress": "10.200.1.10",
                    "IPPrefixLen": 24,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:86:0e:7a:61:0e",
                    "NetworkID": "demo-network",
                    "DriverOpts": null,
                    "IPAMConfig": null,
                    "Links": null
                }
            }
        },
        "ExitCommand": [
            "/usr/bin/podman",
            "--root",
            "/var/lib/containers/storage",
            "--runroot",
            "/var/run/containers/storage",
            "--log-level",
            "error",
            "--cgroup-manager",
            "systemd",
            "--tmpdir",
            "/var/run/libpod",
            "--runtime",
            "runc",
            "--storage-driver",
            "overlay",
            "--storage-opt",
            "overlay.mountopt=nodev,metacopy=on",
            "--events-backend",
            "file",
            "container",
            "cleanup",
            "3562c3f0ccc272e1b5953c8a2bfa192d81b8abe7a611d3a1055cffcaadaf76f5"
        ],
        "Namespace": "",
        "IsInfra": false,
        "Config": {
            "Hostname": "postgresql",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": true,
            "OpenStdin": true,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "TERM=xterm",
                "container=podman",
                "LANG=ja_JP.UTF-8",
                "LC_ALL=ja_JP.UTF-8",
                "HOSTNAME=postgresql",
                "HOME=/root"
            ],
            "Cmd": [
                "/sbin/init"
            ],
            "Image": "localhost/local/c8-postgresql:latest",
            "Volumes": null,
            "WorkingDir": "/",
            "Entrypoint": "",
            "OnBuild": null,
            "Labels": {
                "org.label-schema.build-date": "20200611",
                "org.label-schema.license": "GPLv2",
                "org.label-schema.name": "CentOS Base Image",
                "org.label-schema.schema-version": "1.0",
                "org.label-schema.vendor": "CentOS"
            },
            "Annotations": {
                "io.container.manager": "libpod",
                "io.kubernetes.cri-o.Created": "2020-08-27T11:05:29.112881812+09:00",
                "io.kubernetes.cri-o.TTY": "true",
                "io.podman.annotations.autoremove": "FALSE",
                "io.podman.annotations.init": "FALSE",
                "io.podman.annotations.privileged": "FALSE",
                "io.podman.annotations.publish-all": "FALSE",
                "org.opencontainers.image.stopSignal": "37"
            },
            "StopSignal": 37,
            "CreateCommand": [
                "podman",
                "run",
                "--name",
                "postgresql",
                "--hostname",
                "postgresql",
                "--network",
                "demo-network",
                "--ip",
                "10.200.1.10",
                "-itd",
                "-v",
                "/root/work:/tmp/work",
                "-p",
                "15432:5432",
                "-p",
                "2224:22",
                "local/c8-postgresql"
            ],
            "SystemdMode": true,
            "Umask": "0000",
            "Timeout": 0,
            "StopTimeout": 10
        },
        "HostConfig": {
            "Binds": [
                "/root/work:/tmp/work:rw,rprivate,rbind"
            ],
            "CgroupManager": "systemd",
            "CgroupMode": "host",
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "k8s-file",
                "Config": null,
                "Path": "/var/lib/containers/storage/overlay-containers/3562c3f0ccc272e1b5953c8a2bfa192d81b8abe7a611d3a1055cffcaadaf76f5/userdata/ctr.log",
                "Tag": "",
                "Size": "0B"
            },
            "NetworkMode": "bridge",
            "PortBindings": {
                "22/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "2224"
                    }
                ],
                "5432/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "15432"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": [],
            "CapDrop": [],
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": [],
            "GroupAdd": [],
            "IpcMode": "private",
            "Cgroup": "",
            "Cgroups": "default",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "private",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": [],
            "Tmpfs": {},
            "UTSMode": "private",
            "UsernsMode": "",
            "ShmSize": 65536000,
            "Runtime": "oci",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DiskQuota": 0,
            "KernelMemory": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": 0,
            "OomKillDisable": false,
            "PidsLimit": 2048,
            "Ulimits": [
                {
                    "Name": "RLIMIT_NOFILE",
                    "Soft": 1048576,
                    "Hard": 1048576
                },
                {
                    "Name": "RLIMIT_NPROC",
                    "Soft": 4194304,
                    "Hard": 4194304
                }
            ],
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "CgroupConf": null
        }
    }
]
```

**Comment by rhatdan** [comment]
Could you bring the systemd guys in for this dicussion.

**Comment by mheon** [comment]
Concur - it doesn't sound like Podman is doing anything wrong here. I think we need to ask the systemd team what is going on.

**Comment by rhatdan** [comment]
This should be brought up with systemd, we are sending the correct signal, if systemd is not handling it, then they have a bug or should tell us what is wrong.  When I run ubi8-init
```
 $ podman run -ti  ubi8-init
Resolved "ubi8-init" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull registry.access.redhat.com/ubi8-init:latest...
Getting image source signatures
Checking if image destination supports signatures
Copying blob 06038631a24a done  
Copying blob b9b828ed6e91 done  
Copying blob 262268b65bd5 done  
Copying config 670b0e87da done  
Writing manifest to image destination
Storing signatures
systemd 239 (239-45.el8_4.3) running in system mode. (+PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=legacy)
Detected virtualization container-other.
Detected architecture x86-64.

Welcome to Red Hat Enterprise Linux 8.4 (Ootpa)!

Set hostname to <83d6f2aa2ca8>.
Initializing machine ID from random generator.
Failed to read AF_UNIX datagram queue length, ignoring: No such file or directory
[  OK  ] Listening on Process Core Dump Socket.
[  OK  ] Started Forward Password Requests to Wall Directory Watch.
[  OK  ] Listening on initctl Compatibility Named Pipe.
[  OK  ] Reached target Remote File Systems.
[  OK  ] Listening on Journal Socket (/dev/log).
[  OK  ] Reached target Local File Systems.
[  OK  ] Reached target Swap.
[  OK  ] Listening on Journal Socket.
-.slice: Failed to set memory.min: Operation not permitted
-.slice: Failed to set memory.low: Operation not permitted
-.slice: Failed to set memory.high: Operation not permitted
-.slice: Failed to set memory.max: Operation not permitted
-.slice: Failed to set memory.swap.max: Operation not permitted
-.slice: Failed to set pids.max: Operation not permitted
         Starting Rebuild Dynamic Linker Cache...
         Starting Create System Users...
         Starting Rebuild Journal Catalog...
[  OK  ] Reached target Network is Online.
[  OK  ] Reached target Slices.
         Starting Journal Service...
[  OK  ] Started Dispatch Password Requests to Console Directory Watch.
[  OK  ] Reached target Paths.
[  OK  ] Started Create System Users.
[  OK  ] Started Rebuild Journal Catalog.
[  OK  ] Started Rebuild Dynamic Linker Cache.
         Starting Update is Completed...
[  OK  ] Started Update is Completed.
[  OK  ] Started Journal Service.
         Starting Flush Journal to Persistent Storage...
[  OK  ] Started Flush Journal to Persistent Storage.
         Starting Create Volatile Files and Directories...
[  OK  ] Started Create Volatile Files and Directories.
         Starting Update UTMP about System Boot/Shutdown...
[  OK  ] Started Update UTMP about System Boot/Shutdown.
[  OK  ] Reached target System Initialization.
[  OK  ] Started Daily Cleanup of Temporary Directories.
[  OK  ] Started dnf makecache --timer.
[  OK  ] Reached target Timers.
[  OK  ] Listening on D-Bus System Message Bus Socket.
[  OK  ] Reached target Sockets.
[  OK  ] Reached target Basic System.
[  OK  ] Started D-Bus System Message Bus.
         Starting Permit User Sessions...
[  OK  ] Started Permit User Sessions.
[  OK  ] Reached target Multi-User System.
         Starting Update UTMP about System Runlevel Changes...
[  OK  ] Started Update UTMP about System Runlevel Changes.
```
In another window, I execute
```
$ podman stop -l
```
In the systemd container window I instantly see:
```
[  OK  ] Stopped target Timers.
[  OK  ] Stopped Daily Cleanup of Temporary Directories.
[  OK  ] Stopped target Network is Online.
[  OK  ] Stopped dnf makecache --timer.
         Unmounting /var/log/journal...
[  OK  ] Stopped target Multi-User System.
         Stopping Permit User Sessions...
         Stopping D-Bus System Message Bus...
[FAILED] Failed unmounting /var/log/journal.
[  OK  ] Stopped D-Bus System Message Bus.
[  OK  ] Stopped Permit User Sessions.
[  OK  ] Stopped target Remote File Systems.
[  OK  ] Stopped target Basic System.
[  OK  ] Stopped target Paths.
[  OK  ] Stopped Forward Password Requests to Wall Directory Watch.
[  OK  ] Stopped Dispatch Password Requests to Console Directory Watch.
[  OK  ] Stopped target Sockets.
[  OK  ] Closed D-Bus System Message Bus Socket.
[  OK  ] Stopped target System Initialization.
[  OK  ] Stopped Update is Completed.
[  OK  ] Stopped Rebuild Dynamic Linker Cache.
         Stopping Update UTMP about System Boot/Shutdown...
[  OK  ] Stopped Rebuild Journal Catalog.
[  OK  ] Stopped target Slices.
[  OK  ] Stopped Update UTMP about System Boot/Shutdown.
[  OK  ] Stopped Create Volatile Files and Directories.
[  OK  ] Stopped Create System Users.
[  OK  ] Stopped target Local File Systems.
         Unmounting /run/secrets...
         Unmounting /etc/hosts...
         Unmounting Temporary Directory (/tmp)...
         Unmounting /usr/share/zoneinfo/Etc/UTC...
         Unmounting /run/.containerenv...
         Unmounting /run/lock...
         Unmounting /etc/hostname...
         Unmounting /etc/resolv.conf...
[FAILED] Failed unmounting /run/secrets.
[FAILED] Failed unmounting /etc/hosts.
[FAILED] Failed unmounting Temporary Directory (/tmp).
[FAILED] Failed unmounting /usr/share/zoneinfo/Etc/UTC.
[FAILED] Failed unmounting /run/.containerenv.
[FAILED] Failed unmounting /run/lock.
[FAILED] Failed unmounting /etc/hostname.
[FAILED] Failed unmounting /etc/resolv.conf.
[  OK  ] Reached target Unmount All Filesystems.
[  OK  ] Stopped target Swap.
[  OK  ] Reached target Shutdown.
[  OK  ] Reached target Final Step.
         Starting Halt...
```

**Comment by rhatdan** [comment]
I don't believe Podman has a bug, So closing.

**Comment by knaka888** [comment]
Yes. This is not a bug of Podman.
I checked the shutdown messages by executing "**journalctl -ef**" on CentOS 8 inside a Podman container.
Then I was able to find out the following messages there.

> Sep 24 09:26:33 postgresql systemd[1]: Stopping PostgreSQL 12 database server...
:
> Sep 24 09:26:33 postgresql systemd[1]: Stopped PostgreSQL 12 database server.

I could not find these messages with journalctl commands and /var/log/messages file. Because journald service does not seem to save any shutdown messages like above and old boot messages. I don't know why this happens, but I confirmed that Podman and Systemd are working correctly.
